Week2/homework/maartjes_work.js

```diff
@@ -44,4 +44,23 @@ const tuesday = [
 
 const tasks = monday.concat(tuesday);
 
+
 // Add your code here
+//shortened code
+let paidPerHour = tasks.map(time => time.duration / 60).filter(filteredTime => filteredTime > 2).reduce((accumulator, elm) => accumulator + (elm * 16.50));
+let totalSalary = "€" + paidPerHour.toFixed(2);
+console.log(totalSalary);
+
+//readable code
+//mapped the time per hour 60min
+let salaryPerHour = tasks.map(time => time.duration / 60);
+
+//filtered all durations from 2 hours and above
+let filtered = salaryPerHour.filter(filteredTime => filteredTime > 2);
+
+//reduced the working hours to 16.50 euro per hour
+let reduced = filtered.reduce((accumulator, elm) => accumulator + (elm * 16.50));
```

Comment:
See my comments on naming above.

Now, about the error with the `.reduce` method. The `.reduce()` method takes an optional second parameter: the initial value for the accumulator. If you leave it out it will take the first element of the array to initialise the accumulator and will iterate through the rest of the array. In you case you are multiplying each array element by the hourly rate. However, the first element that was used by `.reduce()` to initialise the accumulator was **not** multiplied by the hourly rate. The end result is that you are underpaying Maartje.

A simple way to correct this is to explicitly call reduce with a second parameter, the constant `0`, to initialise the accumulator. Now, `.reduce()` doesn't take the first array element to initialise the accumulator and iterates through all array elements, including the first one.